### ISSUE TITLE
Opinion : winesapOS should be a flatpak first OS, here is why

### ISSUE THREAD

**Issue by Thijzer** [LukeShortCloud/winesapOS#300]
My opinion is as follows, flatpak mostly,gets the job done. It's becoming transparent to the user. It's installed inside your home directory and it doesn't depend any other system packages to be installed other then flatpak. The flatpak apps can run in isolation and it solve package manager hell.
Also it separates your system from your apps, just like and android phone.
This guarantees a greater stability and less pacman Syu upgrade stress when these to are separated.

Sorry if I repeat myself but I'm a big advocate on this concept, I use docker on a daily basis and without docker my work would a nightmare to maintain. I think that flatpak is the docker of apps, it's almost impossible to break your system when you follow this strictly.

Things like discord, obs, spotify, edge, kodi, protonup, herioc launcher, lutris (soon) are all available as flatpak on flathub.

So my suggestion would be to move some of these applications to their flatpak counterpart and maybe we could look into a post installer that installs these suggestions at first boot?

**Comment by LukeShortCloud** [comment]
I love the idea of switching to Flatpaks. My hope has always been that with the launch of Steam Deck, Flatpaks will continue to improve in stability, performance, and feature parity in comparison with native applications.

Fun fact: I used to work at Red Hat and there I daily drove Fedora Silverblue which is exclusively Flatpaks + (Podman) containers on-top a read-only root file system. I've lived through the joy and pain of that before switching back to Arch Linux and native applications. :-) I still use containers daily. I'm a Sales Engineer at VMware these days. I help showcase the value of our Kubernetes and container technology stack to businesses so I absolutely get it.

I did do a brief experiment with winesapOS a year ago and it wasn't great. I'd be open to trying it out again. Many of our applications are indeed available as Flatpaks as you've pointed out.

Somewhat related: we intend to ship a minimal image with the 3.1.0 release. That would be a good time for us to have a checkbox of applications that users can select to install what they want. That GitHub issue can be found here: https://github.com/LukeShortCloud/winesapOS/issues/254

**Comment by LukeShortCloud** [comment]
Thinking more about about the history of winesapOS, the main reason we haven't gone to Flatpaks is the extra space requirement. It's a lot to have the GNOME and KDE runtime environments (sometimes even different versions of them!) along with apps and their bundled dependencies.

Most of our original users had limited small flash drives. Now our new users (mostly coming in due to us now being a fork of SteamOS 3) are using this on large internal SSDs. Our release images are also huge. The entire release image barely fits into the 28 GiB size we use. The latest build I ran today only had 2 GiB free when it was complete. Our minimal image will partially address the size concern.

The second thing that was holding me back from Flatpaks was the lack of a good Google Chrome Flatpak. From all the Steam Deck coverage, it appears Chrome and Edge finally have decent support on the Deck with development being done to polish it up even more. Things are looking good on this front!

We will definitely adopt Flatpaks one day and probably sooner rather than later.

**Comment by Thijzer** [comment]
Hi Luke,
Thank you very much for your detailed insights here and indeed when you want to target a specific size, every byte counts.
So in line with this information I will set up a new test branch to see if we can move some apps over and check the total size difference by this action. Maybe it helps give a bit more insights in this thought experiment.

**Comment by LukeShortCloud** [comment]
Thanks, that'd be really great to see what your findings are!

**Comment by LukeShortCloud** [comment]
Here's our first issue that would have been avoided if we were using Flatpaks. ;-) https://github.com/LukeShortCloud/winesapOS/issues/312

**Comment by LukeShortCloud** [comment]
Bottles is now our first Flatpak application. I noticed it added about 1 GB to our build for the GNOME runtime and dependencies. Any new applications we add will use Flatpak, if applicable. I imagine adding the KDE Plasma runtime along with other application dependencies may result in another 1-2 GB of space (3 GB total) but that is a complete guess. I'd like to make more optimizations to space and I think the move to Flatpak will force us to do that soon.

**Comment by LukeShortCloud** [comment]
The Google Chrome Flatpaks are now in the stable FlatHub repository as of April 1st (not an April Fool's joke)!

https://flathub.org/apps/details/com.google.Chrome
https://flathub.org/apps/details/com.google.ChromeDev

**Comment by LukeShortCloud** [comment]
I did a test build with the PolyMC Flatpak for https://github.com/LukeShortCloud/winesapOS/issues/314 and that added another 1 GB with KDE Plasma runtime environment, a theme, and other dependencies. That filled up the entire 12 GB that is available on the root partition. Time to do some spring cleaning. I opened up a new GitHub issue to track lowering our storage usage: https://github.com/LukeShortCloud/winesapOS/issues/317

**Comment by LukeShortCloud** [comment]
PolyMC has now replaced MultiMC and is installed via Flatpak: https://github.com/LukeShortCloud/winesapOS/issues/314

**Comment by LukeShortCloud** [comment]
I finally sorted through the pastebin and found most, if not all, of the packages that we can install as Flatpaks.

- Cheese org.gnome.Cheese
- ClamTk com.gitlab.davem.ClamTk
- Discord com.discordapp.Discord
- DosBox-X com.dosbox_x.DOSBox-X
- Gwenview org.kde.gwenview
- KeePassXC org.keepassxc.KeePassXC
- LibreOffice org.libreoffice.LibreOffice
- Lutris net.lutris.Lutris
    - We will wait on installing this as a Flatpak. A separate issue is tracking the upstream progress of it: https://github.com/LukeShortCloud/winesapOS/issues/329
- MangoHud org.freedesktop.Platform.VulkanLayer.MangoHud
- PeaZip io.github.peazip.PeaZip
- Pix org.kde.pix
- Protontricks com.github.Matoking.protontricks
- ProtonUp-Qt net.davidotek.pupgui2
    - Already installed as a Flatpak as of winesapOS 3.0.1.
- Steam com.valvesoftware.Steam
    - We will wait on installing this as a Flatpak. A separate issue is tracking the upstream progress of it: https://github.com/LukeShortCloud/winesapOS/issues/328
- VLC org.videolan.VLC.Plugin.makemkv

**Comment by LukeShortCloud** [comment]
Flatseal com.github.tchx84.Flatseal is also an application I keep hearing about that makes tweaking Flatpaks easy. It seems very essential. We will install that, too.

**Comment by LukeShortCloud** [comment]
@Thijzer You will be happy to know that we have converted the majority of our packages over to Flatpaks! Here are the remaining packages that are pending upstream fixes to their Flatpaks along with our GitHub issues tracking the progress of each one. We will include those packages in a later update. I suspect that with the Steam Deck out in the wild, those Flatpaks with remaining issues will have everything fixed soon.

- [Firefox org.mozilla.firefox](https://github.com/LukeShortCloud/winesapOS/issues/334)
- [Heroic Games Launcher com.heroicgameslauncher.hgl](https://github.com/LukeShortCloud/winesapOS/issues/333)
- [Lutris net.lutris.Lutris](https://github.com/LukeShortCloud/winesapOS/issues/329)
- [MangoHud org.freedesktop.Platform.VulkanLayer.MangoHud](https://github.com/LukeShortCloud/winesapOS/issues/336)
- [Steam com.valvesoftware.Steam](https://github.com/LukeShortCloud/winesapOS/issues/328)

**Comment by LukeShortCloud** [comment]
@Thijzer Our stable winesapOS 3.1.0 is out now with Flatpaks as our preferred application instalation method. Thanks for the suggestion in the first place! We would not have made the switch as soon as we did otherwise. I'm very glad we did make the switch. 🙂

https://github.com/LukeShortCloud/winesapOS/releases/tag/3.1.0
